javalang/tokenizer.py

```diff
@@ -289,7 +290,7 @@ def read_decimal_float_or_integer(self):
 
         self.read_decimal_integer()
 
-        if self.data[self.j] not in '.eEfFdD':
+        if self.j >= len(self.data) or self.data[self.j] not in '.eEfFdD':
```

Comment:
Could you add an equivalent check to `read_hex_integer_or_float`?

Reply:
Done.